src/plugins/routing.js

```diff
@@ -27,6 +27,8 @@ function localePathFactory (i18nPath, routerPath) {
     // Match route without prefix for default locale
     if (locale === defaultLocale && STRATEGY === STRATEGIES.PREFIX_AND_DEFAULT) {
       name += routesNameSeparator + defaultLocaleRouteNameSuffix
+    } else if (STRATEGY === STRATEGIES.NO_PREFIX) {
```

Comment:
Nit: Sorry but can you instead use ternary expression on the line that declares `name`. I find the flow a bit weird when variable is assigned something just to be overridden later like that.

Maybe like this:
```suggestion
    let name = route.name + (STRATEGY === STRATEGIES.NO_PREFIX ? '' : routesNameSeparator + locale)
```